sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.time.Duration;
+
+/**
+ * Represents the connection config with {@link ConnectionMode#DIRECT} associated with Cosmos Client in the Azure Cosmos DB database service.
+ */
+public final class DirectConnectionConfig {
```

Comment:
in Direct mode we still establish some connections to GW (master resource operations and query plan retrieval, etc). 
How are we controlling the GW Connection specifics in Direct mode?

Reply:
Good point, we are only allowing for default values configurations for `Gateway` connection properties in case of `Direct` connection mode. 
Something to think about, I will give it a thought.

Reply:
@moderakh - I have exposed these gateway client properties to `CosmosClientBuilder`
So that end users can also set these even in case of direct connection mode. 
```
private Duration requestTimeoutGateway = DEFAULT_REQUEST_TIMEOUT;
private Duration idleConnectionTimeoutGateway = DEFAULT_IDLE_CONNECTION_TIMEOUT;
private int maxConnectionPoolSizeGateway = DEFAULT_MAX_POOL_SIZE;
```